packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageViewWrapper.kt

```diff
@@ -170,13 +171,32 @@ class ExpoImageViewWrapper(context: Context, appContext: AppContext) : ExpoView(
 
   fun setIsAnimating(setAnimating: Boolean) {
     val resource = activeView.drawable
+    /* Animatable animations always start from the beginning when resumed.
+    So we check first if the resource is a GifDrawable, because it can continue
+    from where it was paused. */
+    when (resource) {
+      is GifDrawable -> toggleGifDrawableAnimation(resource, setAnimating)
+      is Animatable -> toggleAnimatableAnimation(resource, setAnimating)
+    }
+  }
 
-    if (resource is Animatable) {
-      if (setAnimating) {
-        resource.start()
+  private fun toggleGifDrawableAnimation(resource: GifDrawable, setAnimating: Boolean) {
```

Comment:
```suggestion
  private fun setIsAnimating(resource: GifDrawable, setAnimating: Boolean) {
```
It is not toggling anything when you're passing `setAnimating` as an option.
Also, we should be able to use function overloading rather than creating a separate function for each type.